src/data/sum/basic.lean

```diff
@@ -228,13 +211,126 @@ swap_swap
 @[simp] lemma swap_right_inverse : function.right_inverse (@swap α β) swap :=
 swap_swap
 
+section lift_rel
+
+/-- Lifts pointwise two relations between `α` and `γ` and between `β` and `δ`to a relation between
+`α ⊕ β` and `γ ⊕ δ`. -/
+inductive lift_rel (r : α → γ → Prop) (s : β → δ → Prop) : α ⊕ β → γ ⊕ δ → Prop
+| inl {a c} : r a c → lift_rel (inl a) (inl c)
+| inr {b d} : s b d → lift_rel (inr b) (inr d)
+
+variables {r r₁ r₂ : α → γ → Prop} {s s₁ s₂ : β → δ → Prop} {a : α} {b : β} {c : γ} {d : δ}
+  {x : α ⊕ β} {y : γ ⊕ δ}
+
+lemma lift_rel_inl_inl : lift_rel r s (inl a) (inl c) ↔ r a c :=
+⟨λ h, by { cases h, assumption }, lift_rel.inl⟩
+
+lemma lift_rel_inl_inr : ¬ lift_rel r s (inl a) (inr d) .
+
+lemma lift_rel_inr_inl : ¬ lift_rel r s (inr b) (inl c) .
+
+lemma lift_rel_inr_inr : lift_rel r s (inr b) (inr d) ↔ s b d :=
+⟨λ h, by { cases h, assumption }, lift_rel.inr⟩
+
+instance [Π a c, decidable (r a c)] [Π b d, decidable (s b d)] :
+  Π (ab : α ⊕ β) (cd : γ ⊕ δ), decidable (lift_rel r s ab cd)
+| (inl a) (inl c) := decidable_of_iff' _ lift_rel_inl_inl
+| (inl a) (inr d) := decidable.is_false lift_rel_inl_inr
+| (inr b) (inl c) := decidable.is_false lift_rel_inr_inl
+| (inr b) (inr d) := decidable_of_iff' _ lift_rel_inr_inr
+
+lemma lift_rel.mono (hr : ∀ a b, r₁ a b → r₂ a b) (hs : ∀ a b, s₁ a b → s₂ a b)
+  (h : lift_rel r₁ s₁ x y) :
+  lift_rel r₂ s₂ x y :=
```

Comment:
Is it worth having a version stating this as
```suggestion
lemma lift_rel.mono' (hr : r₁≤ r₂) (hs : s₁≤ s₂) : lift_rel r₁ s₁ ≤ lift_rel r₂ s₂ :=
```
The two are defeq.

Reply:
I don't know. I did that in a PR a while ago and people told me nobody was using the order on relations.